### PR TITLE
Refactor and improve animation logic.

### PR DESCRIPTION
**Pull request by nicksay**
- Make the start state more reliable by delaying for 17ms
  (1 frame at 60 frames/second) before running the animation.
- Refactor the animation logic into separate dedicated functions for
  clarity: prepare, run, complete.
- Store animation-related data in a dedicated object.
- Rename a couple DOM methods to better communicate what they do.

### CODE REVIEW

**Comment by nicksay** [comment]
I'm going to split the refactoring and logic changes out from the demo CSS changes so they're isolated.

**Comment by nicksay** [comment]
Updated this PR to remove the changes to the development demo

**Comment by nicksay** [comment]
The demo CSS changes are now in #296.

File: src/client/nav/response.js [hunk 1]
```diff
@@ -220,122 +221,82 @@ spf.nav.response.process = function(url, response, opt_info, opt_callback) {
...
           } else {
             el.innerHTML = extracted.html;
-            spf.debug.debug('    body update', id);
             installScripts();
           }
+
         } else {
-          spf.tasks.suspend(key);  // Suspend main queue for animation.
-          var animationKey = spf.tasks.key(el);
+
```
Comment (awbraunstein) [inline src/client/nav/response.js]: extra nl here?

File: src/client/nav/response.js [hunk 2]
```diff
@@ -220,122 +221,82 @@ spf.nav.response.process = function(url, response, opt_info, opt_callback) {
...
-            // TODO(nicksay): Consider using a sub-queue for JS execution.
-            spf.tasks.suspend(key);
-            spf.nav.response.installScripts_(extracted, function() {
-              spf.debug.debug('    body js', id);
-              spf.tasks.resume(key, sync);  // Resume main queue after JS.
-              spf.debug.debug('  process task done: body', id);
-            });
-          };
-          var innerHtmlHandler = /** @type {Function} */(
+
```
Comment (awbraunstein) [inline src/client/nav/response.js]: extra nl here?

File: src/client/nav/response.js [hunk 3]
```diff
@@ -220,122 +221,82 @@ spf.nav.response.process = function(url, response, opt_info, opt_callback) {
...
-              spf.tasks.resume(key, sync);  // Resume queue after handler.
+              // Resume main queue after the experimental HTML handler.
+              spf.tasks.resume(key, sync);
             });
           } else {
             el.innerHTML = extracted.html;
-            spf.debug.debug('    body update', id);
             installScripts();
           }
+
```
Comment (awbraunstein) [inline src/client/nav/response.js]: extra nl here

File: src/client/nav/response.js [hunk 4]
```diff
@@ -220,122 +221,82 @@ spf.nav.response.process = function(url, response, opt_info, opt_callback) {
...
+          // Resume main queue after animation is done.
+          spf.tasks.add(
+              animation.key,
+              spf.bind(function() {
+                installScripts();
+                spf.tasks.resume(key, sync);
+              }, null),
+              0);
+          spf.tasks.run(animation.key);
+
```
Comment (awbraunstein) [inline src/client/nav/response.js]: extra nl here

File: src/client/nav/response.js (hunk 2 above)
Comment (nicksay) [inline src/client/nav/response.js]: Removed

File: src/client/nav/response.js (hunk 3 above)
Comment (nicksay) [inline src/client/nav/response.js]: Removed

File: src/client/nav/response.js (hunk 1 above)
Comment (nicksay) [inline src/client/nav/response.js]: Removed

File: src/client/nav/response.js (hunk 4 above)
Comment (nicksay) [inline src/client/nav/response.js]: Removed

**Comment by nicksay** [comment]
Extra blank lines removed.

**Comment by awbraunstein** [comment]
lgtm

**Comment by nicksay** [comment]
Post-associating this with #299.

**Pull request merged**